### ISSUE TITLE
Enum data-source binds to specific assembly version

### ISSUE THREAD

**Issue by robertjf** [leekelleher/umbraco-contentment#197]
### Which Contentment version are you using?

3.0.1

### Which Umbraco version are you using? For example: 8.14.1 - don't just write v8

9.3.0-rc

### Bug summary

In our devops ci/cd pipeline we have a script that updates the assembly version prior to building.  Contentment is looking for a specific version in the Enum specification as follows:

```
{"dataSource":[{"key":"Umbraco.Community.Contentment.DataEditors.EnumDataListSource, Umbraco.Community.Contentment","value":{"enumType":["LifeSupportsCounselling.Core, Version=1.0.0.2239, Culture=neutral, PublicKeyToken=null","LifeSupportsCounselling.Core.BlockPartitionStyle"]}}],"listEditor":[{"key":"Umbraco.Community.Contentment.DataEditors.TemplatedListDataListEditor, Umbraco.Community.Contentment","value":{"template":"<div style=\"border: solid 2px rgb(228,231,247);border-radius:0.5rem;padding: 1rem;width: 30vw;margin: 1rem auto;background-color: #fff;\">\n<h3 ng-bind=\"item.name\"></h3>\n<img ng-if=\"item.value !== 'None'\" ng-src=\"/assets/images/pickers/border-style/{{item.value}}.png\" />\n</div>\n\n","allowClear":"0","htmlAttributes":[]}}]}
```

Because the dll is compiled with an updated version number, this results in the following errors in the logs:

```
Feb 9, 2022 11:46:19 AM | Error | dw0sdwk000048 | Unable to load target type.
-- | -- | -- | --
System.IO.FileNotFoundException: Could not load file or assembly 'LifeSupportsCounselling.Core, Version=1.0.0.2211, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified. File name: 'LifeSupportsCounselling.Core, Version=1.0.0.2211, Culture=neutral, PublicKeyToken=null'    at 
System.Reflection.RuntimeAssembly.InternalLoad(ObjectHandleOnStack assemblyName, ObjectHandleOnStack requestingAssembly, StackCrawlMarkHandle stackMark, Boolean throwOnFileNotFound, ObjectHandleOnStack assemblyLoadContext, ObjectHandleOnStack retAssembly)    at 
System.Reflection.RuntimeAssembly.InternalLoad(AssemblyName assemblyName, RuntimeAssembly requestingAssembly, StackCrawlMark& stackMark, Boolean throwOnFileNotFound, AssemblyLoadContext assemblyLoadContext)    at 
System.Reflection.Assembly.Load(String assemblyString)    at 
Umbraco.Community.Contentment.DataEditors.EnumDataListSource.GetValueType(Dictionary`2 config)
```

This impacts not only the rendering of the page, but also the ability to edit the Enum Data List Source configuration.

### Steps to reproduce

1. Create a Data Type with an Enum Data Source based on an Enum in an assembly
2. Update the version number in the Enum assembly and recompile.

### Do you have Umbraco ModelsBuilder enabled?

- [X] Yes, it is enabled.

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Microsoft Edge

**Comment by leekelleher** [comment]
@robertjf Thanks for reporting this. Yikes! 😱

How in earth am I gonna fix this?! 😬
Open to suggestions, _(if anyone else is following along at home?)_ 😅

**Comment by leekelleher** [comment]
@robertjf Good news, after investigating this, I'm able to load the assembly without the additional version & culture bits.
Fixed in 96a2c471c64aa649f4159f4d8ed1f488646dbb26, it'll be part of upcoming v3.1.0 release, _(I'm aiming for next week)_.

**Comment by robertjf** [comment]
Hey @leekelleher thanks for the speedy attention and fix!  My project and I thank you :)  Looking forward to testing the resolution :)
